### ISSUE TITLE
Your title here

### ISSUE THREAD

**Issue by Cube-Controls** [thingsboard/thingsboard#10164]
**Component**

 * Generic

**Description**
Frist time testing  TB with ESP32, I'm getting "[TB] Unable to serialize json data" on any  tb.sendTelemetryData or tb.sendAttributeData.
tb.sendTelemetryJson is working.

**Environment**

 * OS:  Windows 11
 * ThingsBoard: 0.12.2 Arduino

**Comment by gittimotap** [comment]
Hi. I'm having the exact same issue. I'm using Firebeetle 2 ESP32 and no matter which code example I take, once I post anything to the server, I get the response "[TB] Unable to serialize json data". I'm able to use the API via command line using MQTT or CURL with no issues. For reproducing you can use demo.thingsboard.io and just send data, it will answer with the same error.

**Comment by Cube-Controls** [comment]
I think problem is in Arduino SDK , something with functions that create JSON.
`tb.sendTelemetryJson` function is working ok , because you can create JSON manualy before sending.

Functions that require SDK to create JSON don't work.
Maybe this is wrong repository I posted question, probably should be in [thingsboard-client-sdk] repository.

**Comment by gittimotap** [comment]
I'm using 2 different boards, https://wiki.dfrobot.com/SKU_DFR0868_Beetle_ESP32_C3 and https://wiki.dfrobot.com/FireBeetle_Board_ESP32_E_SKU_DFR0654, both give me the same issue when posting anything to the server. Do you mean this https://github.com/espressif/arduino-esp32 for Arduino SDK?

**Comment by gittimotap** [comment]
Seems like the issue is with ArduinoJSON library, 6.21.5 works, I've opened a bug for this

**Comment by Cube-Controls** [comment]
No there is SDK from thingsboard https://github.com/thingsboard/thingsboard-client-sdk that is used for ESP32.

tb.sendAttributeData and tb.sendTelemetryData are implemented there and it looks that problem is with serialization of data to JSON before sending to thingsboard server. It could be problem and with one of the libraries that are used for json serialization.


This is not working and causing "[TB] Unable to serialize json data"

```

    tb.sendTelemetryData("temperature", random(10, 20));
    tb.sendAttributeData("rssi", WiFi.RSSI());
```

If you serialize data in the code and send josn directly using tb.sendTelemetryJson then it is working.

```
void sendDataToThingsBoard(float temp, int hum) {
  String jsonData = "{\"temperature\":" + String(temp) + ", \"humidity\":" + String(hum) + "}";
  tb.sendTelemetryJson(jsonData.c_str());
  Serial.println("Data sent");
}

void sendAttributesToThingsBoard(void) {
  String jsonData = "{\"rsss\":" + String(1234) + ", \"test\":" + String(456) + "}";
  tb.sendAttributeJSON(jsonData.c_str());
  Serial.println("Attributes sent");
}
```

I just hoped someone already figured out this issue, I will try to debug to see what is going on.

**Comment by Cube-Controls** [comment]
> Seems like the issue is with ArduinoJSON library, 6.21.5 works, I've opened a bug for this

I confirmed that, I will close this issue.

**Comment by utkarshjosh** [comment]
Yes, downgrading from ^7~ to 6.21.5 really worked. Thanks A lot. :partying_face:

**Comment by gittimotap** [comment]
7.0.4 also works now
